app/models/work_packages/scopes/relatable.rb

```diff
@@ -40,33 +40,114 @@ module Relatable
       #   are simply somehow related, such relations do not follow the "non circular" nor the "ancestor/descendant" rule.
       # * Relation::TYPE_PARENT: Since creating a new relationship will remove the old parent relationship, current ancestors
       #   (except the direct parent) are relatable to. Descendants however are not since that would create a circle.
+      #   In addition to the existing hierarchy, the FOLLOWS relationships are taken into account. Predecessors and successors
+      #   (FOLLOWS relationship) may also not be related to via a PARENT relation. However, parents and children of those
+      #   predecessors/successors can be related to. But they still need to be considered in the code since they might be
+      #   part of an almost completed circle which would be closed if a work package is added as a parent. E.g. in
+      #     WP1 --- follows ---> WP2 --- parent ---> WP3 --- follows ---> WP4
+      #   WP4 is not a valid parent candidate whereas WP3 is.
+      #   Work packages related via follows/precedes to any descendant of a work package are exempt from being relatable right
+      #   away as it would create a circle.
+      #
+      # The implementation focuses on excluding candidates. It does so in two parts:
+      #   * Excluding all work packages with which a direct relation already exist.
+      #   * Excluding work packages that are related transitively.
+      #
+      # The first is straightforward. The second is more complicated and also depends on the type of relation that is
+      # queried for. It uses a CTE to recursively find all work packages with which a transitive relationship of interest based
+      # on the rules outlined above exist.
+      #
+      # For most of the relation types, this includes work packages in the direction towards or away from the
+      # work package to be related to the hierarchy (ancestors and descendants - not siblings and aunts).
+      # Note that only one of the directions is of interest, which is the one opposite to the one that is queried for via the
+      # type. This is to prevent a circle of relationships. For the 'related' type, only the ancestors and descendants are of
+      # interest.
       #
-      # The implementation right away excludes all work packages with which a direct relation already exists
-      # and uses a CTE to find all the work packages with which a transitive relationship based on the rules outlined above
-      # exist.
+      # For FOLLOWS/PRECEDES relationships both directions need to be taken into account, and again, this includes those of the
+      # hierarchy. This prevents creating relations in a structure like this:
+      #     queried WP
+      #       --- follows ---> Predecessor WP
+      #       --- parent ---> Predecessor's parent WP
+      #       <-- follows --- Predecessor's parent's successor WP
```

Comment:
I have a hard time reading this kind of schema.
Should it be expanded like this?
```
      #      queried WP --- follows ---> Predecessor WP
      #      queried WP --- parent ---> Predecessor's parent WP
      #      queried WP <-- follows --- Predecessor's parent's successor WP
```

Maybe it's hard for me because when reading `A --- follows ---> B`, here A is on the left and B on the right, but on a timeline B would be on the left and A on the right.

Reply:
I can try to pick up your notation and see if that is easier for the two of us to understand. But I fear that the direction change will be inevitable at least for some of the invalid trees.

Reply:
Updated the docs in a29cf73bb2ea3ba08594b53051c37312639b1465